src/pages/iv-opening-cost/Forms/OpeningCostForm.js

```diff
@@ -0,0 +1,163 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import { ControlContext } from 'src/providers/ControlContext'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+
+export default function OpeningCostForm({ labels, maxAccess, record }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const recordId = String(record?.year) + String(record?.itemId) || null;
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.OpeningCost.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      itemId: null,
+      year: '',
+      avgCost: 0
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      year: yup.number().required(),
+      itemId: yup.string().required(),
+      avgCost: yup.number().min(0).max(999999999)
```

Comment:
can we have a record without cost? if not please add required here and for the field because i can remove the 0 from the field and submit without a value for this field

Reply:
in asp we cannot I will add it